src/frontmatter.ts

```diff
@@ -0,0 +1,306 @@
+            return;
+          }
+
+          const source = await readFile(file, 'utf8');
+          const {data: frontmatter} = matter(source);
+          allFrontMatter.push({
+            ...(frontmatter as FrontMatter),
+            slug: formatSlug(fileName),
+            sourcePath: path.join('docs', fileName),
+          });
+        },
+        {concurrency: FILE_CONCURRENCY_LIMIT}
+      )
+    )
+  );
+
+  // Add API documentation pages (categories and endpoints)
+  const categories = await apiCategories();
+  categories.forEach(category => {
+    allFrontMatter.push({
+      title: category.name,
+      slug: `api/${category.slug}`,
+    });
+
+    category.apis.forEach(api => {
+      allFrontMatter.push({
+        title: api.name,
+        slug: `api/${category.slug}/${api.slug}`,
+      });
+    });
+  });
+
+  // Now process all common files for each platform and expand them into platform/guide pages
+  const platformsPath = path.join(docsPath, 'platforms');
+  for await (const platform of await opendir(platformsPath)) {
+    if (platform.isFile()) {
+      continue;
+    }
+    const platformName = platform.name;
+
+    let platformFrontmatter: PlatformConfig = {};
+    const configPath = path.join(platformsPath, platformName, 'config.yml');
+    try {
+      platformFrontmatter = yaml.load(
+        await readFile(configPath, 'utf8')
+      ) as PlatformConfig;
+    } catch (err) {
+      // the file may not exist and that's fine, for anything else we throw
+      if (err.code !== 'ENOENT') {
+        throw err;
+      }
+    }
+
+    const commonPath = path.join(platformsPath, platformName, 'common');
+    try {
+      await access(commonPath);
+    } catch (err) {
+      continue;
+    }
+
+    const commonFileNames: string[] = (await getAllFilesRecursively(commonPath)).filter(
+      p => path.extname(p) === '.mdx'
+    );
+
+    const commonFiles = await Promise.all(
+      commonFileNames.map(
+        limitFunction(
+          async commonFileName => {
+            const source = await readFile(commonFileName, 'utf8');
+            const {data: frontmatter} = matter(source);
+            return {commonFileName, frontmatter: frontmatter as FrontMatter};
+          },
+          {concurrency: FILE_CONCURRENCY_LIMIT}
+        )
+      )
+    );
+
+    // Add common files for the platform itself (no guide)
+    await Promise.all(
+      commonFiles.map(
+        limitFunction(
+          async f => {
+            if (!isSupported(f.frontmatter, platformName)) {
+              return;
+            }
+
+            const subpath = f.commonFileName.slice(commonPath.length + 1);
+            const slug = f.commonFileName
+              .slice(docsPath.length + 1)
+              .replace(/\/common\//, '/');
+            const noFrontMatter = (
+              await Promise.allSettled([
+                access(path.join(docsPath, slug)),
+                access(path.join(docsPath, slug.replace('/index.mdx', '.mdx'))),
```

Comment:
access checks are pretty expensive. We've already computed the existing file paths upfront (via `getAllFilesRecursively`), can we just re-use that?

Reply:
fixed in [f253fa5](https://github.com/getsentry/sentry-docs/pull/15452/commits/f253fa56f1de18a8241fd96086b986ae8e9f1f1a)